bundles/action/org.openhab.action.twitter/src/main/java/org/openhab/action/twitter/internal/Twitter.java

```diff
@@ -58,34 +64,130 @@ public static boolean sendTweet(@ParamDoc(name = "tweetTxt", text = "the Tweet t
             return false;
         }
 
+        return true;
+    }
+
+    /**
+     * Internal method for sending a tweet, with or without image
+     * 
+     * @param tweetTxt
+     *            text string to be sent as a Tweet
+     * @param fileToAttach
+     *            the file to attach. May be null if no attached file.
+     * 
+     * @return <code>true</code>, if sending the tweet has been successful and
+     *         <code>false</code> in all other cases.
+     */
+    private static boolean doSendTweet(final String tweetTxt, final File fileToAttach) {
```

Comment:
Can't be renamed that way as it would generate a signature collision. Moreover is a private internal method, not to be used in rules.

Reply:
Of course it can. You just have to cast your nulls appropriately:

Line 114:

````
return sendTweet(tweetTxt, (File)null);
````

Another option would be to make the File parameter optional so that you can avoid sending null altogether.